sdk/src/main/scala/io/horizen/account/state/AccountStateView.scala

```diff
@@ -131,11 +134,11 @@ class AccountStateView(
     metadataStorageView.getForgerBlockCounters
   }
 
-  def resetForgerPoolAndBlockCounters(consensusEpochNumber: ConsensusEpochNumber): Unit = {
+  def subtractForgerPoolBalanceAndResetBlockCounters(consensusEpochNumber: ConsensusEpochNumber, poolBalanceDistributed: BigInteger): Unit = {
     if (Version1_2_0Fork.get(consensusEpochNumber).active) {
       val forgerPoolBalance = getBalance(WellKnownAddresses.FORGER_POOL_RECIPIENT_ADDRESS)
```

Comment:
Yes, makes sense - I added a check for that that throws exception - this will fail fast so to inform about any bad modifications in the future. Thanks!